src/main/kotlin/com/craftschemers/hub/minigame/Minigame.kt

```diff
@@ -1,3 +1,18 @@
 package com.craftschemers.hub.minigame
 
-class Minigame
+import com.craftschemers.hub.HubPlayer
+
+abstract class Minigame(val name: String, val type: GameType) {
+    var state = GameState.LOBBY
+    val players = mutableListOf<HubPlayer>()
+
+    fun addPlayer(player: HubPlayer) {
+        players += player
+        player.game = type
+    }
+
+    fun hasPlayer(player: HubPlayer): Boolean = player in players
+
+    abstract fun start()
```

Comment:
Do we also want a `stop()` function or something similar?

Reply:
we will likely utilize the [`GameState` enum](https://github.com/craft-schemers/hub-server/blob/joining-and-leaving/src/main/kotlin/com/craftschemers/hub/minigame/GameState.kt) instead. not super important for this PR